docs/stateful-resources.md

```diff
@@ -0,0 +1,104 @@
+
+> Avoid changing the logical ID of a resource after it has been created. Since AWS CloudFormation identifies resources by their logical ID, if you change the logical ID of a resource, AWS CloudFormation deletes the existing resource, and then creates a new resource with the new logical ID, which may cause service interruption or data loss.
+
+### Property updates
+When updating the properties of some [resources](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-template-resource-type-ref.html), AWS is unable to perform the update in place. This is indicated by the "Update requires" section of the property definition, and further described [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-update-behaviors.html).
+
+In most instances, replacement is expected:
+- [S3 bucket name](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket.html#cfn-s3-bucket-name)
+- [DynamoDB table key schema](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-keyschema)
+
+In some instances, replacement is expected but not destructive:
+- [EC2 launch configuration user data](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-autoscaling-launchconfiguration.html#cfn-autoscaling-launchconfiguration-userdata)
+
+In some instances, replacement is unexpected:
+- [Security group description](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-security-group.html#cfn-ec2-securitygroup-groupdescription)
+- [Application load balancer type](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticloadbalancingv2-loadbalancer.html#cfn-elasticloadbalancingv2-loadbalancer-type)
+
+## What can be done
+At a minimum, CD of your infrastructure with RIff-Raff should be configured. To provide a shorter feedback loop, the following can be used.
+
+### Logical ID
+When authoring a stack using GuCDK every resource will be added to a `GuStack`:
+
+```typescript
+class MyStack extends GuStack {
+  constructor(app: App, id: string, props: GuStackProps) {
+    super(app, id, props);
+
+    // resources for my stack
+    new Bucket(this, "DataBucket");
+  }
+}
+```
+
+When `MyStack` is synthesized into a CloudFormation template, we'll have a resource with a logical ID of `DataBucket<GUID>` as AWS CDK attempts to create uniqueness.
+
+When wanting to preserve the logical ID for a resource, one can call `overrideLogicalId` on `GuStack`:
+
+```typescript
+class MyStack extends GuStack {
+  constructor(app: App, id: string, props: GuStackProps) {
+    super(app, id, props);
+
+    // resources for my stack
+    const dataBucket = new Bucket(this, "DataBucket");
+    this.overrideLogicalId(dataBucket, { logicalId: "DataBucket", reason: "Retaining a stateful resource previously defined in YAML" });
```

Comment:
```suggestion
    this.overrideLogicalId(dataBucket, { 
      logicalId: "DataBucket", 
      reason: "Retaining a stateful resource previously defined in YAML" 
   });
```

Sorry to be a linter here, but this just means that it will all display nicely on a standard screen size without needing to scroll.

Reply:
Great point! Thanks! I've become so used to the linter running on save 😅 .